dashboard/app/helpers/levels_helper.rb

```diff
@@ -277,11 +282,36 @@ def blockly_options(local_assigns={})
     end
 
     # Set some values that Blockly expects on the root of its options string
-    app_options = {'levelId' => @level.level_num}
-    app_options['scriptId'] = @script.id if @script
-    app_options['levelGameName'] = @level.game.name if @level.game
-    app_options['scrollbars'] = blockly_value(@level.scrollbars) if @level.is_a?(Blockly) && @level.scrollbars
-    [level, app_options]
+    app_options = {
+      baseUrl: "#{ActionController::Base.asset_host}/blockly/",
+      app: @game.try(:app),
+      levelId: @level.level_num,
+      level: level,
+      callouts: @callouts,
+      cacheBust: blockly_cache_bust,
+      autoplayVideo: @autoplay_video_info,
+      report: {
+          fallback_response: @fallback_response,
+          callback: @callback,
+      },
+    }
+    app_options[:scriptId] = @script.id if @script
+    app_options[:levelGameName] = @level.game.name if @level.game
+    app_options[:scrollbars] = blockly_value(@level.scrollbars) if @level.is_a?(Blockly) && @level.scrollbars
+    app_options[:skinId] = @level.skin if @level.is_a?(Blockly)
+    app_options[:level_source_id] = @level_source_id if @level_source_id
+    app_options[:sendToPhone] = request.location.try(:country_code) == 'US' ||
+        (!Rails.env.production? && request.location.try(:country_code) == 'RD') if request
```

Comment:
Looks like 'RD' is for reserved IP addresses: http://stackoverflow.com/questions/13691815/ipinfodb-lookup-returns-countrycode-rd.  My guess is so that this feature works when testing on localhost?